ofrak_core/test_ofrak/components/test_llm_components.py

```diff
@@ -0,0 +1,71 @@
+import asyncio
+import os
+
+import pytest
+
+import ofrak_ghidra
+import test_ofrak
+from ofrak import OFRAKContext
+from ofrak.core.llm import LlmAnalyzer, LlmAnalyzerConfig, LlmProgramAnalyzer, LlmAttributes
+
+
+@pytest.fixture()
+@pytest.mark.asyncio
+async def ollama():
+    async def run_task():
+        proc = None
+        try:
+            proc = await asyncio.subprocess.create_subprocess_exec("ollama", "serve")
+            await proc.communicate()
+        except asyncio.CancelledError:
+            if proc is not None:
+                proc.kill()
+
+    task = asyncio.create_task(run_task())
+    await asyncio.sleep(5)
+    yield
+    task.cancel()
+
+
+@pytest.fixture()
+@pytest.mark.asyncio
+async def model(ollama) -> str:
+    # Smallest chat model currently available on ollama
+    model_name = "qwen2.5:0.5b"
+    proc = await asyncio.subprocess.create_subprocess_exec("ollama", "pull", model_name)
+    await proc.communicate()
+    yield model_name
+
+
+@pytest.fixture(autouse=True)
+def ghidra_components(ofrak_injector):
+    ofrak_injector.discover(ofrak_ghidra)
+
+
+async def test_llm_component(ofrak_context: OFRAKContext, model: str):
```

Comment:
Can we use the https://github.com/redballoonsecurity/ofrak/blob/master/ofrak_core/test_ofrak/unit/component/analyzer/analyzer_test_case.py#L34 to run these tests?

Right now these tests are not validating that the analyzer successfully generates the needed attributes.

Reply:
I can do that validation in these tests. It does not make sense to use `AnalyzerTests` subclass here since the analyzer must be run with `resource.run`. It requires a config, and thus cannot be called with `resource.analyze`, as happens in `AnalyzerTests` methods.

Reply:
Also the tests do call `get_attributes` at the end, so they're validating that the attributes are added.